core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -276,24 +291,24 @@ private[spark] object CoarseGrainedExecutorBackend extends Logging {
           // scalastyle:off println
           System.err.println(s"Unrecognized options: ${tail.mkString(" ")}")
           // scalastyle:on println
-          printUsageAndExit()
+          printUsageAndExit(classNameForEntry)
```

Comment:
You want `classNameForEntry.stripSuffix("$")`, either here or in the caller, or the help message will be wrong.

Reply:
Nice finding! I may still think many things in Java way. Will address.